compair/tests/api/test_answers.py

```diff
@@ -924,8 +953,19 @@ def test_get_user_answers(self):
             # test unsaved query
             rv = self.client.get(url, query_string={'unsaved': True})
             self.assert200(rv)
+            # The answer API compares timestamps ("modified" vs "created") to determine if an answer is "unsaved".
```

Comment:
Can you actually modify the answer modal and endpoint so it uses `modified == created` and `draft == true`. This was a situation I didn't handle/catch when adding it (if a new non-draft answer is added somehow, it shouldn't be unsaved)

Thanks!

Reply:
@andrew-gardener  The API supports bool parameters `unsaved` and `draft`.  If the model's `saved` property is changed to depend on `draft` property, it will be contradictory to invoke the API with `unsaved` without `draft`.

So, will change the logic to skip `draft` filter if  `unsaved` is specified.

Does it make sense?

&nbsp; | unsaved | unsaved not specified
--- | --- | ---
**draft** | not Answer.saved | Answer.draft = True
**draft not specified** | not Answer.saved | Answer.draft = False

The model will be like:
```
    @hybrid_property
    def saved(self):
        return not (self.modified == self.created and self.draft)

    @saved.expression
    def saved(cls):
        return not_(and_(cls.modified == cls.created, cls.draft))
```

And API will be like:
```
        if params.get('unsaved'):
            query = query.filter(not_(Answer.saved))
        else:
            # move the draft parameter checking here
            query = query.filter_by(draft=params.get('draft'))
```

Reply:
`unsaved` param is used only as workaround for getting uuids to work with xAPI (need to uuid to exist before the answer would normally). Its currently only called in one place and I want to try to find a better way of handling it in the future. 

I don't think its worth building to much around it (I think you can just use the this part):
```
        if params.get('unsaved'):
            query = query.filter(not_(Answer.saved))
```
for the endpoint

Also I prefer not using negation around multiple argurments if possible (makes it harder to follow)
```
    @hybrid_property
    def saved(self):
        return self.modified != self.created or not self.draft

    @saved.expression
    def saved(cls):
        return or_(cls.modified != cls.created, cls.draft == False)
```